### ISSUE TITLE
Add OpenZFS support 

### ISSUE THREAD

**Issue by zeekus** [aws/aws-parallelcluster#3899]
-AWS ParallelCluster version [ 3.1.2 ]
-cluster config: see bluefishtestfsx.yaml
-Clustername: testami cli errored out before build started.
-Output of `pcluster describe-cluster` N/A 

Bug and description: 
1. created an OpenZFS filesystem in the web interface using the smallest available size (64GB). 
2. configured yaml to use the new 'Shared Storage' area. See run details. It appears the cli is giving a non descriptive error as if it is encountering output it doesn't expect. 
```
(apc-ve) myuser@myserver:$ pcluster version
{
  "version": "3.1.2"
}

(apc-ve) myuser@myserver:$ pcluster create-cluster --cluster-configuration ./bluefishtestfsx.yaml --cluster-name testami --region us-east-1
{
  "message": "'NoneType' object has no attribute 'get'"
}


#filename: bluefishtestfsx.yaml
Region: us-east-1
Image:
  Os: centos7
  CustomAmi: ami-0849b3cca0757b921
SharedStorage:
  - Name: modelingtest
    StorageType: FsxLustre
    MountDir: /fsx
    FsxLustreSettings:
      FileSystemId: fs-045ffe08c17984010
HeadNode:
  InstanceType: c4.8xlarge
  Networking:
    SubnetId: subnet-a04625d7
    AdditionalSecurityGroups:
    - sg-06d8a61ae120cc35e
  Ssh:
    KeyName: mykey
  LocalStorage:
    RootVolume:
      Size: 100
  Iam:
    AdditionalIamPolicies:
      - Policy: arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
      - Policy: arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy
    S3Access:
      - BucketName: '*'
      - EnableWriteAccess: true
        BucketName: parallelcluster
Scheduling:
  Scheduler: slurm
  SlurmQueues:
    - Name: compute
      ComputeResources:
        - Name: slurmworkers
          InstanceType: c4.8xlarge
          MinCount: 0
          MaxCount: 8
      Iam:
        AdditionalIamPolicies:
          - Policy: arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
          - Policy: arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy
        S3Access:
          - BucketName: '*'
          - EnableWriteAccess: true
            BucketName: parallelcluster
      Networking:
        SubnetIds:
          - subnet-a04625d7
        PlacementGroup:
          Enabled: true
      ComputeSettings:
        LocalStorage:
          RootVolume:
            Size: 40
```

**Comment by francesco-giordano** [comment]
Hi @zeekus thank you for your interest,
currently FSx for OpenZFS is not supported as SharedStorage. FSx for Lustre is the supported alternative.

https://docs.aws.amazon.com/parallelcluster/latest/ug/SharedStorage-v3.html

However we take, as action item, the improvement of the validator message.

Thanks

**Comment by enrico-usai** [comment]
Hi @zeekus, we're working on adding support for existing OpenZFS file systems: https://github.com/aws/aws-parallelcluster/pull/4048

**Comment by gmarciani** [comment]
Support for OpenZFS included in [ParallelCluster 3.2.0](https://github.com/aws/aws-parallelcluster/releases/tag/v3.2.0)
